packages/integration-tests/tests/social-session.test.ts

```diff
@@ -0,0 +1,121 @@
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { mockSocialConnectorId, mockSocialConnectorConfig } from '@/__mocks__/connectors-mock';
+import {
+  signInWithSocial,
+  getAuthWithSocial,
+  registerWithSocial,
+  bindWithSocial,
+  signInWithUsernameAndPassword,
+} from '@/api';
+import MockClient from '@/client';
+import { setUpConnector, createUserByAdmin } from '@/helpers';
+import { generateUsername, generatePassword } from '@/utils';
+
+const state = 'foo_state';
+const redirectUri = 'http://foo.dev/callback';
+const code = 'auth_code_foo';
+
+describe('social sign-in and register', () => {
+  const userId = crypto.randomUUID();
+
+  beforeAll(async () => {
+    await setUpConnector(mockSocialConnectorId, mockSocialConnectorConfig);
+  });
+
+  it('register with social', async () => {
+    const client = new MockClient();
+
+    await client.initSession();
+    assert(client.interactionCookie, new Error('Session not found'));
+
+    await expect(
+      signInWithSocial(
+        { state, connectorId: mockSocialConnectorId, redirectUri },
+        client.interactionCookie
+      )
+    ).resolves.toBeTruthy();
+
+    const response = await getAuthWithSocial(
+      { connectorId: mockSocialConnectorId, data: { state, redirectUri, code, userId } },
+      client.interactionCookie
+    ).catch((error: unknown) => error);
+
+    // User with social does not exist
+    expect(response instanceof HTTPError && response.response.statusCode === 422).toBe(true);
```

Comment:
Should be `.toBeTruthy()`?

Reply:
in this case could strictly equal to true as it is a bool type